pkg/provider/provider.go

```diff
@@ -119,9 +120,11 @@ func streamInterceptor(srv interface{}, ss grpc.ServerStream,
 	info *grpc.StreamServerInfo, handler grpc.StreamHandler) (err error) {
 	err = handler(srv, ss)
 	if err != nil {
-		// modified for wrong file downloads
-		// to not zap errors
-		if storage.ErrKeyNotFound.Has(err) {
+		// no zap errors for canceled or wrong file downloads
+		if storage.ErrKeyNotFound.Has(err) ||
+			status.Code(err) == codes.Canceled ||
+			status.Code(err) == codes.Unavailable ||
```

Comment:
Although, shouldn't these be filtered at a different level? Do we want to ignore all Canceled and Unavailable errors everywhere?

Reply:
I don't find a single place where we use these codes in our code base. I think they come from the gRPC layer.

I think the idea of the interceptor is to just log unexpected errors, not filter all the rest entirely. If an error is not logged, it is still returned up the stack.

Anywhere, if you have a better idea how to handle this, I am all ears.

Reply:
I don't think there's a better approach, unless we add separate logging to each peer. So, yeah, lets go with this approach.